test/types/helpers.test-d.ts

```diff
@@ -129,14 +128,77 @@ expectError(
 
     for await (const response of scrollSearch) {
       expectAssignable<ScrollSearchResponse>(response)
-      expectType<ResponseBody<Record<string, any>>>(response.body)
+      expectType<Record<string, any>>(response.body)
       expectType<unknown[]>(response.documents)
       expectType<unknown>(response.meta.context)
     }
   }
 }
 
 // with type defs
+{  
+  interface ShardsResponse {
+    total: number;
+    successful: number;
+    failed: number;
+    skipped: number;
+  }
+  
+  interface Explanation {
+    value: number;
+    description: string;
+    details: Explanation[];
+  }
+  
+  interface SearchResponse<T> {
+    took: number;
+    timed_out: boolean;
+    _scroll_id?: string;
+    _shards: ShardsResponse;
+    hits: {
+      total: number;
+      max_score: number;
+      hits: Array<{
+        _index: string;
+        _type: string;
+        _id: string;
+        _score: number;
+        _source: T;
+        _version?: number;
+        _explanation?: Explanation;
+        fields?: any;
+        highlight?: any;
+        inner_hits?: any;
+        matched_queries?: string[];
+        sort?: string[];
+      }>;
+    };
+    aggregations?: any;
+  }
+  
+  interface Source {
+    foo: string
+  }
+
+  async function test () {
+    const scrollSearch = client.helpers.scrollSearch<Source, SearchResponse<Source>>({
```

Comment:
shouldn't it follow the same type order as other generics? `<ResponseBody, RequestBody, Context, Document>`?

Reply:
The reason why I put `TDocument` as the first parameter, is because the `scrollSearch` returns `AsyncIterable<ScrollSearchResponse<TDocument, TResponse, TContext>>`, and `ScrollSearchResponse` is defined as follows:
```ts
export interface ScrollSearchResponse<TDocument = unknown, TResponse = Record<string, any>, TContext = unknown> extends ApiResponse<TResponse, TContext> {
  clear: () => Promise<void>
  documents: TDocument[]
}
```
Having `TDocument` as first parameter would allow users to access directly `response.documents` instead of accessing the full search response.